receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_scraper.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cpuscraper
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/shirou/gopsutil/cpu"
+	"go.opencensus.io/trace"
+
+	"github.com/open-telemetry/opentelemetry-collector/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/pdata"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/pdatautil"
+	"github.com/open-telemetry/opentelemetry-collector/internal/data"
+)
+
+// Scraper for CPU Metrics
+type Scraper struct {
+	config    *Config
+	consumer  consumer.MetricsConsumer
+	startTime time.Time
+}
+
+// NewCPUScraper creates a set of CPU related metrics
+func NewCPUScraper(ctx context.Context, cfg *Config, consumer consumer.MetricsConsumer) (*Scraper, error) {
+	return &Scraper{config: cfg, consumer: consumer}, nil
+}
+
+// Start
+func (c *Scraper) Start(ctx context.Context) error {
+	go func() {
+		c.startTime = time.Now()
+
+		ticker := time.NewTicker(c.config.CollectionInterval())
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				c.scrapeMetrics(ctx)
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	return nil
+}
+
+// Close
+func (c *Scraper) Close(ctx context.Context) error {
+	return nil
+}
+
+func (c *Scraper) scrapeMetrics(ctx context.Context) {
+	ctx, span := trace.StartSpan(ctx, "cpuscraper.scrapeMetrics")
+	defer span.End()
+
+	metricData := data.NewMetricData()
+	metrics := pdatautil.InitializeMetricSlice(metricData)
```

Comment:
You probably need to populate the resource informations about the host. See https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions#host

Maybe an issue.

Reply:
@jrcamp - there's a comment about resources in the [Host Metrics](https://docs.google.com/document/d/1i-QG-TxkIWVbdx86UM2GXcFvpeCV7zSBsPmMyMkOEBk/edit) spec document, but I wasn't 100% sure which direction we want to go based on that.

Does it make more sense to populate resource info in the receiver or leave it empty and expect that a processor is configured to append resource info (which would be applied to all receivers)?

Intuitively it seems like we should always record relevant resource info when we create metrics (i.e. in the receiver in this case), and then the resourceprocessor can be used to override this if desired.

Reply:
Yeah it does seem weird but I'm not sure I see a better way given the architecture of the collector.

@bogdandrutu for host.* and cloud.* labels we will need these added to resources emitted by other non-host metrics receivers (e.g. a redis receiver, etc.). It seems to consistently apply these labe;ls it needs to be done in a processor so that all receivers can receive the same resource labels for host.* and cloud.*. Should hostmetrics also collect these as resources itself? If it does then does it share code with this new processor? 

I agree it does seem weird to emit an empty resource here but given the decoupled nature of the collector it seems most consistent to apply them in a processor.

Reply:
I think we can start with having a library that determines these resources, then we can split it in a processor or shared it between every receiver that needs these library - I don't see any concern of having every receiver using the same shared library and directly append the resource, it is not like we cause too much pressure to an api like in the case of kubernetes cluster metadata, so not worried if we actually fetch these data 10 times when we start the collector.

Anyway we need this mini-library that determines the host metadata so we can start by implementing it here.

Reply:
I prefer that approach as well; I don't really like the idea of creating metrics without resource info and relying on users to configure the collector appropriately. So to make sure I'm on the same page as you guys, the plan would be to:

1. For any receiver that generates its own metrics, automatically detect the host resource and assign this info to metrics at creation time 
2. Unrelatedly, if the user wants to replace / append the resource info with static info, they can use the existing resource processor

Re #1, I guess we should put this library in a separate repo (since it could be useful for the Go SDK as well?), and this would be based off of the existing OCensus library: https://github.com/census-ecosystem/opencensus-go-resource (updated to use Otel apis, etc)

I put this into a ticket: https://github.com/open-telemetry/opentelemetry-collector/issues/871 - feel free to comment/edit

Reply:
>Re 1, I guess we should put this library in a separate repo (since it could be useful for the Go SDK as well?), and this would be based off of the existing OCensus library: https://github.com/census-ecosystem/opencensus-go-resource (updated to use Otel apis, etc)

Don't overthink let's start it in this repo for the moment then see how do we share with opentelemetry go SDK :)

Reply:
You can also start a separate PR for that :)

Reply:
Yup, I'll probably come back to this one in a few weeks